src/libmongoc/tests/bsonutil/bson-val.c

```diff
@@ -45,9 +47,16 @@ bson_val_from_value (const bson_value_t *value)
    bson_init (&tmp);
    bson_append_value (&tmp, "v", 1, value);
    as_string = bson_as_canonical_extended_json (&tmp, NULL);
-   /* This produces: { "v" : {...} }. Strip off the wrapping "v" and braces. */
-   val->as_string = bson_strndup (as_string + 7, strlen (as_string) - 9);
-   bson_free (as_string);
+
+   if (!as_string) {
```

Comment:
Interesting, did this occur in tests? I think `bson_as_canonical_extended_json` only returns `NULL` if the document is malformed.

Reply:
Yes, it showed up as a seg fault in testing.  I guess we were running a malformed document through!  Returning `val` without a string still caused a test failure, but one that was much easier to debug.  I can't remember which test was causing this now, though.